sdk/lib/opentelemetry/sdk/trace/export/batch_span_processor.rb

```diff
@@ -177,6 +177,9 @@ def reset_on_fork(restart_thread: true)
             @pid = pid
             spans.clear
             @thread = restart_thread ? Thread.new { work } : nil
+          rescue ThreadError => e
+            @metrics_reporter.add_to_counter('otel.bsp.error', labels: { 'reason' => 'ThreadError' })
+            OpenTelemetry.logger.error "OpenTelemetry error in BatchSpanProcessor#reset_on_fork #{e.message}"
```

Comment:
Are we supposed use `OpenTelemetry.logger.error` or `OpenTelemetry.handle_error` for this purpose (I'm unsure).

Reply:
Great question - I can't think of a good reason *not* to use `OpenTelemetry.handle_error`.

Reply:
Updated, I think it makes sense to let the users decide on how to handle this error.  And it has a sane default of logging it.